napari/layers/labels/labels.py

```diff
@@ -410,6 +410,12 @@ def seed(self, seed):
         self.refresh()
         self.events.selected_label()
 
+    @_ImageBase.colormap.setter
+    def colormap(self, colormap):
+        super()._set_colormap(colormap)
+        self._selected_color = self.get_color(self.selected_label)
+        self.refresh()
```

Comment:
```suggestion
    @colormap.setter
    def colormap(self, colormap):
        self._colormap = ensure_colormap(colormap)
        self._selected_color = self.get_color(self.selected_label)
        self._update_thumbnail()
        self.events.colormap()
```

I wonder if i wouldn't prefer this as an implementation to avoid the second refresh? we also wouldn't need to change the image layer at all here. it's maybe a bit more duplicated code, so i can see why not. thoughts @jni?

Reply:
@sofroniewn That doesn't work — you get a "colormap is not defined" error. Hence the workaround above.

Reply:
Yup good catch - you need the `_ImageBase` there (I updated comment above so it should work). (or you have to add the whole property in, but that feels like a lot of duplication). You have to add the `ensure_colormap` import in too, but you don't have to add the `_set_colormap` function to the mixin, which feels a little circuitous and you avoid the 2nd refresh.

Thoughts? If you still prefer `_set_colormap` and don't mind the 2nd refresh then it's ok

Reply:
I'm neutral on whether I prefer adding `_set_colormap` versus implementing it here.

But, the second `refresh()` _is_ required. Trying Nick's suggestion "looks" good with my example in this PR, but won't update when you change the colors.  (e.g. `viewer.layers[0].colormap = napari.utils.colormaps.label_colormap(num_colors=131)`).

Reply:
BTW, I don't think we *should* have to call `refresh()`, but it's clear that `self.events.colormap()` does not lead to a refresh of the colorbox. I suspect that would be easy to do, but I don't know where.

Reply:
nice, yeah - i think it would be drop `self.refresh()` entirely - is that possible now @perlman?

Reply:
Removed; the colorbox now calls `update()` when when it receives a colormap event.